src/compartment.c

```diff
@@ -436,28 +544,48 @@ comp_map(struct Compartment* to_map)
     for (size_t i = 0; i < to_map->seg_count; ++i)
     {
         curr_seg = to_map->segs[i];
-        if (curr_seg->file_sz == curr_seg->mem_sz)
+        map_result = mmap((void*) curr_seg->mem_bot,
+                                curr_seg->mem_sz,
+                                /*curr_seg->prot_flags,*/ // TODO currently need read/write to inject the intercepts, consider better option
+                                PROT_READ | PROT_WRITE | PROT_EXEC,
+                                MAP_PRIVATE | MAP_FIXED | MAP_ANONYMOUS,
+                                -1, 0);
+        if (map_result == MAP_FAILED)
         {
-            map_result = mmap((void*) curr_seg->mem_bot,
-                                    curr_seg->file_sz,
-                                    /*curr_seg->prot_flags,*/ // TODO currently need read/write to inject the intercepts, consider better option
-                                    PROT_READ | PROT_WRITE | PROT_EXEC,
-                                    MAP_PRIVATE | MAP_FIXED,
-                                    to_map->fd, curr_seg->offset);
+            perror("Error mapping comp segment");
+            exit(1);
         }
-        else
+        printf("HERE COMP\n");
```

Comment:
Debug `printf`?

Reply:
I think this is means to be https://github.com/capablevms/CHERI-ELF-comp/pull/13/commits/6a5003d9598b32b1f5bbb8572cf94b82a9c9f111